packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -607,3 +608,176 @@ class _AnimatedDefaultTextStyleState extends AnimatedWidgetBaseState<AnimatedDef
     );
   }
 }
+
+/// Specifies which of the children to show. See [AnimatedCrossFade].
+///
+/// The child that is shown will fade in, and while the other will fade out.
+enum CrossFadeState {
+  /// A state where the first child is shown and the second child is completely
+  /// invisible.
+  showFirst,
+  /// A state where the second child is shown and the first child is completely
+  /// invisible.
+  showSecond
+}
+
+/// A widget that cross-fades between two children and animates between their
+/// sizes. The animation is controlled through the [crossFadeState] parameter.
```

Comment:
It animates its own size between their sizes, no? Their size doesn't actually animate.